frontend/src/routes/Governance/common/util.tsx

```diff
@@ -463,30 +464,74 @@ export function getPolicySetPolicies(policies: Policy[], policySet: PolicySet) {
   )
 }
 
-export function getPolicyRemediation(policy: Policy | undefined) {
+export function getPolicyRemediation(policy: Policy | undefined, propagatedPolicies: Policy[]) {
   if (!policy) {
     return ''
   }
-  if (policy.spec?.remediationAction) {
-    return policy.spec.remediationAction
-  }
   const templates = policy.spec['policy-templates']
+  const rootRA = policy.spec.remediationAction
+  if (rootRA === REMEDIATION_ACTION.ENFORCE) {
+    return rootRA
+  }
+  // NOTE RA means Remediation Action
   let remediationAggregation = '-'
-  templates?.forEach((template: PolicyTemplate) => {
-    const templateRemediation = template.objectDefinition.spec.remediationAction
-    if (remediationAggregation === 'inform/enforce') {
-      return
-    } else if (remediationAggregation !== '-' && remediationAggregation !== templateRemediation) {
-      remediationAggregation = 'inform/enforce'
-      return
-    } else if (remediationAggregation !== templateRemediation && templateRemediation) {
-      remediationAggregation = templateRemediation
-      return
+  propagatedPolicies.forEach((propaPolicy) => {
+    const propaPolicyRA = propaPolicy.spec.remediationAction
+    if (propaPolicyRA !== rootRA) {
+      if (remediationAggregation === '-') {
+        remediationAggregation = REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
+      } else if (!remediationAggregation.includes(REMEDIATION_ACTION.ENFORCE_OVERRIDDEN)) {
+        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
+      }
+    } else if (propaPolicyRA != undefined) {
+      if (remediationAggregation === '-') {
+        remediationAggregation = REMEDIATION_ACTION.INFORM
+      } else if (remediationAggregation == REMEDIATION_ACTION.ENFORCE_OVERRIDDEN) {
+        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
+      }
     }
   })
+
+  let remediationAggregationTemp = '-'
+
+  if (!rootRA) {
```

Comment:
I tried to avoid additional procedure when root remediation is not null, we don't need to calculate policy-template when we have  root remediation

Reply:
@yiraeChristineKim I think you misunderstood my comment. Here is what I meant:

```typescript
  const rootRA = policy.spec.remediationAction
  if (rootRA === REMEDIATION_ACTION.ENFORCE) {
    return rootRA
  } else if (!rootRA) {
    let remediationAggregationTemp = '-'

    templates?.forEach((template: PolicyTemplate) => {
      const templateRemediation = template.objectDefinition.spec.remediationAction
      if (remediationAggregationTemp === REMEDIATION_ACTION.INFORM_ENFORCE) {
        return
      } else if (remediationAggregationTemp !== '-' && remediationAggregationTemp !== templateRemediation) {
        remediationAggregationTemp = REMEDIATION_ACTION.INFORM_ENFORCE
      } else if (remediationAggregationTemp !== templateRemediation && templateRemediation) {
        remediationAggregationTemp = templateRemediation
      }
    })

    if (remediationAggregationTemp === '-') {
      return remediationAggregationTemp
    }

    rootRA = remediationAggregationTemp
}


propagatedPolicies.forEach((propaPolicy) => {
....
```

Reply:
What about this case, root:null A: enforce B: enforce, template[0]: enforce template[1]: enforce, we expect this case `enforce`, but this logic result is enforce override

Reply:
Oh also root:null A: enforce B: null, template[0]: inform template[1]: inform,  we expect `inforce/enforce overridden `but actual result is `enforce overridden`

Reply:
Case root: null, clusterA: enforce, clusterB: null, tmp[0] enforce  tmp[1] enforce, we expect: inform/enfoce (overridden), but actual is enforce (overridden)

we need to change   propagatedPolicies.forEach((propaPolicy) => {

Reply:
That example above should be `enforce` because the root policy has no remediation action and both templates are `enforce`, so it's the equivalent of the root policy having `remediationAction: enforce`.

Reply:
I just unit test with the logic the result is ` inform/enforce (overridden)`

Reply:
@yiraeChristineKim are you able to debug a bit to see why?

Reply:
```
  propagatedPolicies.forEach((propaPolicy) => {
    const propaPolicyRA = propaPolicy.spec.remediationAction
    if (propaPolicyRA !== rootRA) {
      if (remediationAggregation === '-') {
        remediationAggregation = REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
      } else if (!remediationAggregation.includes(REMEDIATION_ACTION.ENFORCE_OVERRIDDEN)) {
        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
      }
    } else if (propaPolicyRA != undefined) {
      if (remediationAggregation === '-') {
        remediationAggregation = REMEDIATION_ACTION.INFORM
      } else if (remediationAggregation == REMEDIATION_ACTION.ENFORCE_OVERRIDDEN) {
        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
      }
    }
```
enforce == enforce same so it goes to `else if `.  remediationAggregation = inform. The ext turn is undefined and enforce so these two don't match, it falls to  
```
else if (!remediationAggregation.includes(REMEDIATION_ACTION.ENFORCE_OVERRIDDEN)) {
        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
      }
```

Reply:
Another tests also failed

Reply:
This is what I propose:
```diff
diff --git a/frontend/src/routes/Governance/common/util.test.tsx b/frontend/src/routes/Governance/common/util.test.tsx
index 66e0f7fc..4ef8a654 100644
--- a/frontend/src/routes/Governance/common/util.test.tsx
+++ b/frontend/src/routes/Governance/common/util.test.tsx
@@ -256,7 +256,7 @@ describe('Test getPolicyRemediation', () => {
       ]
       rootPolicy.spec['policy-templates'] = rootPolicyTemplates
       expect(getPolicyRemediation(rootPolicy, [clusterA, clusterB, clusterC])).toEqual(
-        REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
+        REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
       )
     })
     test('getPolicyRemediation should be inform/enforce when template-remediations mixed', () => {
@@ -284,7 +284,7 @@ describe('Test getPolicyRemediation', () => {
       ]
       rootPolicy.spec['policy-templates'] = rootPolicyTemplates
       expect(getPolicyRemediation(rootPolicy, [clusterA, clusterB, clusterC])).toEqual(
-        REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
+        REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
       )
     })
     test('getPolicyRemediation should be enforce when all template-remediations = enforce', () => {
diff --git a/frontend/src/routes/Governance/common/util.tsx b/frontend/src/routes/Governance/common/util.tsx
index 8980ed64..048bdf89 100644
--- a/frontend/src/routes/Governance/common/util.tsx
+++ b/frontend/src/routes/Governance/common/util.tsx
@@ -468,59 +468,64 @@ export function getPolicyRemediation(policy: Policy | undefined, propagatedPolic
   if (!policy) {
     return ''
   }
+
   const templates = policy.spec['policy-templates']
-  const rootRA = policy.spec.remediationAction
+  let rootRA = policy.spec.remediationAction
   if (rootRA === REMEDIATION_ACTION.ENFORCE) {
     return rootRA
   }
-  // NOTE RA means Remediation Action
-  let remediationAggregation = '-'
-  propagatedPolicies.forEach((propaPolicy) => {
-    const propaPolicyRA = propaPolicy.spec.remediationAction
-    if (propaPolicyRA !== rootRA) {
-      if (remediationAggregation === '-') {
-        remediationAggregation = REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
-      } else if (!remediationAggregation.includes(REMEDIATION_ACTION.ENFORCE_OVERRIDDEN)) {
-        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
-      }
-    } else if (propaPolicyRA != undefined) {
-      if (remediationAggregation === '-') {
-        remediationAggregation = REMEDIATION_ACTION.INFORM
-      } else if (remediationAggregation == REMEDIATION_ACTION.ENFORCE_OVERRIDDEN) {
-        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
-      }
-    }
-  })
-
-  let remediationAggregationTemp = '-'
 
   if (!rootRA) {
     templates?.forEach((template: PolicyTemplate) => {
       const templateRemediation = template.objectDefinition.spec.remediationAction
-      if (remediationAggregationTemp === REMEDIATION_ACTION.INFORM_ENFORCE) {
+
+      if (!templateRemediation) {
         return
-      } else if (remediationAggregationTemp !== '-' && remediationAggregationTemp !== templateRemediation) {
-        remediationAggregationTemp = REMEDIATION_ACTION.INFORM_ENFORCE
-      } else if (remediationAggregationTemp !== templateRemediation && templateRemediation) {
-        remediationAggregationTemp = templateRemediation
+      }
+
+      if (rootRA === REMEDIATION_ACTION.INFORM_ENFORCE) {
+        return
+      }
+
+      if (!rootRA) {
+        rootRA = templateRemediation
+      } else if (rootRA !== templateRemediation) {
+        rootRA = REMEDIATION_ACTION.INFORM_ENFORCE
       }
     })
+  }
+
+  let remediationOverride = ''
+  let allOverridden = propagatedPolicies.length !== 0
+  propagatedPolicies.forEach((propaPolicy) => {
+    const propaPolicyRA = propaPolicy.spec.remediationAction
+    if (!propaPolicyRA) {
+      allOverridden = false
 
-    const notOverriden = propagatedPolicies.every((p) => p.spec.remediationAction == undefined)
-    if (notOverriden) {
-      remediationAggregation = remediationAggregationTemp
+      return
     }
 
-    if (remediationAggregation === REMEDIATION_ACTION.ENFORCE_OVERRIDDEN) {
-      if (remediationAggregationTemp === REMEDIATION_ACTION.ENFORCE) {
-        remediationAggregation = REMEDIATION_ACTION.ENFORCE
-      } else {
-        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
-      }
+    if (propaPolicyRA === rootRA) {
+      allOverridden = false
+
+      return
     }
+
+    if (!remediationOverride) {
+      remediationOverride = REMEDIATION_ACTION.ENFORCE
+    }
+  })
+
+  if (allOverridden) {
+    return REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
+  }
+
+  // No overrides are present
+  if (!remediationOverride) {
+    return rootRA
   }
 
-  return remediationAggregation
+  return REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
 }
 
 export function getPropagatedPolicies(rootPolicy: Policy, propaPolicies: Policy[]) {
```


For an easier view, this is the function code I propose:

```typescript
export function getPolicyRemediation(policy: Policy | undefined, propagatedPolicies: Policy[]) {
  if (!policy) {
    return ''
  }

  const templates = policy.spec['policy-templates']
  let rootRA = policy.spec.remediationAction
  if (rootRA === REMEDIATION_ACTION.ENFORCE) {
    return rootRA
  }

  if (!rootRA) {
    templates?.forEach((template: PolicyTemplate) => {
      const templateRemediation = template.objectDefinition.spec.remediationAction

      if (!templateRemediation) {
        return
      }

      if (rootRA === REMEDIATION_ACTION.INFORM_ENFORCE) {
        return
      }

      if (!rootRA) {
        rootRA = templateRemediation
      } else if (rootRA !== templateRemediation) {
        rootRA = REMEDIATION_ACTION.INFORM_ENFORCE
      }
    })
  }

  let remediationOverride = ''
  let allOverridden = propagatedPolicies.length !== 0
  propagatedPolicies.forEach((propaPolicy) => {
    const propaPolicyRA = propaPolicy.spec.remediationAction
    if (!propaPolicyRA) {
      allOverridden = false

      return
    }

    if (propaPolicyRA === rootRA) {
      allOverridden = false

      return
    }

    if (!remediationOverride) {
      remediationOverride = REMEDIATION_ACTION.ENFORCE
    }
  })

  if (allOverridden) {
    return REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
  }

  // No overrides are present
  if (!remediationOverride) {
    return rootRA
  }

  return REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
}
```